x/bsonx/bsoncore/document.go

```diff
@@ -261,32 +264,57 @@ func (d Document) DebugString() string {
 // String outputs an ExtendedJSON version of Document. If the document is not valid, this method
 // returns an empty string.
 func (d Document) String() string {
-	if len(d) < 5 {
+	return d.StringN(math.MaxInt)
+}
+
+// StringN stringifies a document upto N bytes
+func (d Document) StringN(n int) string {
+	if len(d) < 5 || n <= 0 {
 		return ""
 	}
+
 	var buf strings.Builder
-	buf.WriteByte('{')
 
-	length, rem, _ := ReadLength(d) // We know we have enough bytes to read the length
+	buf.WriteByte('{')
 
+	length, rem, _ := ReadLength(d)
 	length -= 4
 
 	var elem Element
 	var ok bool
+
 	first := true
-	for length > 1 {
-		if !first {
-			buf.WriteByte(',')
-		}
-		elem, rem, ok = ReadElement(rem)
-		length -= int32(len(elem))
-		if !ok {
-			return ""
+	truncated := false
+
+	if n > 0 {
+		for length > 1 {
+			if !first {
+				buf.WriteByte(',')
+			}
+			elem, rem, ok = ReadElement(rem)
+			length -= int32(len(elem))
+			if !ok {
+				return ""
+			}
+
+			// TODO: Performance Improvement
```

Comment:
Remove this comment or clarify the performance improvement associated with truncating the element at this point.